src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -285,21 +285,35 @@ private static unsafe int IndexOfHtmlEncodingChars(string s, int startPos)
 
         #region UrlEncode implementation
         
-        private static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
+        private unsafe static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
         {
+            Debug.Assert(originalBytes != null);
```

Comment:
> I'd really prefer that these checks were explicit and not compiled out in Retail builds.

OK; should I do something like throw an `ArgumentNullException` if it's null in both Debug/Release instead, then?

> I'd also just say to use checked arithmetic when doing operations which might overflow (instead of checking after the fact).

Good point, I'd forgotten that it's possible for these sorts of things to overflow; will add some logic to protect against this.